src_kt/javatests/arcs/storage/RamDiskBackingStoreIntegrationTest.kt

```diff
@@ -0,0 +1,117 @@
+import arcs.storage.driver.RamDiskDriverProvider
+import arcs.storage.driver.RamDiskStorageKey
+import com.google.common.truth.Truth.assertThat
+import com.google.common.truth.Truth.assertWithMessage
+import kotlinx.atomicfu.atomic
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.After
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+/** Tests for [RamDiskDriver] coupled with [BackingStore]. */
+@ExperimentalCoroutinesApi
+@RunWith(JUnit4::class)
+class RamDiskBackingStoreIntegrationTest {
+  private lateinit var ramDiskProvider: DriverProvider
+
+  @Before
+  fun setup() {
+    ramDiskProvider = RamDiskDriverProvider()
+  }
+
+  @After
+  fun teardown() {
+    RamDisk.clear()
+    DriverFactory.clearRegistrationsForTesting()
+  }
+
+  @Suppress("UNCHECKED_CAST")
+  @Test
+  fun allowsStorageOf_aNumberOfObjects() = runBlockingTest {
+    val storageKey = RamDiskStorageKey("unique")
+    val baseStore = Store<CrdtCount.Data, CrdtCount.Operation, Int>(
+      StoreOptions(storageKey, ExistenceCriteria.ShouldCreate, CountType(), StorageMode.Backing)
+    )
+    val store = baseStore.activate() as BackingStore
+
+    val count1 = CrdtCount()
+    count1.applyOperation(Increment("me", version = 0 to 1))
+
+    val count2 = CrdtCount()
+    count2.applyOperation(MultiIncrement("them", version = 0 to 10, delta = 15))
+
+    val id = store.on(MultiplexedProxyCallback { _, _ -> true })
+    assertThat(store.onProxyMessage(ProxyMessage.ModelUpdate(count1.data, id), "thing0")).isTrue()
+    assertThat(store.onProxyMessage(ProxyMessage.ModelUpdate(count2.data, id), "thing1")).isTrue()
+
+    store.idle()
+
+    val message = atomic<ProxyMessage<CrdtCount.Data, CrdtCount.Operation, Int>?>(null)
+    val muxId = atomic<String?>(null)
+    var job = Job()
+
+    val id2 = store.on(MultiplexedProxyCallback { m, eventId ->
+      message.value = m as ProxyMessage<CrdtCount.Data, CrdtCount.Operation, Int>
+      muxId.value = eventId
+      job.complete()
+      return@MultiplexedProxyCallback true
+    })
+
+    store.onProxyMessage(ProxyMessage.SyncRequest(id2), "thing0")
+    job.join()
+    message.value.assertHasData(count1)
+    assertThat(muxId.value ?: "huh, it was null.").isEqualTo("thing0")
+
+    message.value = null
+    muxId.value = null
+    job = Job()
+    store.onProxyMessage(ProxyMessage.SyncRequest(id2), "thing1")
+    job.join()
+    message.value.assertHasData(count2)
+    assertThat(muxId.value ?: "huh, it was null.").isEqualTo("thing1")
+
+    message.value = null
+    muxId.value = null
+    job = Job()
+    store.onProxyMessage(ProxyMessage.SyncRequest(id2), "not-a-thing")
+    job.join()
+    message.value.assertHasData(CrdtCount())
+    assertThat(muxId.value ?: "huh, it was null.").isEqualTo("not-a-thing")
```

Comment:
Is the `?: "huh, it was null."` needed? You can't do `assertThat(null)`?

Reply:
I tried that first, but for some reason the compiler was borking on that...